clients/client/src/main/java/org/projectnessie/client/StreamingUtil.java

```diff
@@ -94,27 +118,32 @@ private StreamingUtil() {
    *
    * @param api The {@link NessieApiV1} to use
    * @param ref a named reference (branch or tag name) or a commit-hash
+   * @param builderCustomizer a Consumer of the {@link GetCommitLogBuilder} that is passed to the
+   *     caller of this method. It allows the caller to customize the GetCommitLogBuilder with
+   *     additional parameters (e.g.: hashOnRef, untilHash, filter).
+   * @param maxRecords - a maximum number of records in the stream. If it is {@code
+   *     Optional.empty()} the stream will be unbounded.
+   * @param fetchAdditionalInfo - specify whether it should fetch {@link FetchOption#ALL} or {@link
+   *     FetchOption#MINIMAL}
    * @return stream of {@link CommitMeta} objects
    */
   public static Stream<LogEntry> getCommitLogStream(
       @NotNull NessieApiV1 api,
-      @NotNull String ref,
-      @Nullable String hashOnRef,
-      @Nullable String untilHash,
-      @Nullable String filter,
-      OptionalInt maxRecords,
+      String ref,
+      @NotNull Consumer<GetCommitLogBuilder> builderCustomizer,
+      @NotNull OptionalInt maxRecords,
       boolean fetchAdditionalInfo)
```

Comment:
It's fine to remove `fetchAdditionalInfo`

Reply:
done